### PR TITLE
fix: readme and readme-dev fixes and grooming

### PR DESCRIPTION
**Pull request by barriebyron**
This PR does the first-level editorial grooming for the welcome sign of all open-source projects: the readme files!

Explain your changes:
* Removes incorrect newsletter signup link (to Coda!) and sends readers to https://docs.minaprotocol.com/participate/online-communities and website home page to sign up for newsletter
* Add missing links and fix typos in readme and readme-dev
* Contributing updates
* Edits for clarity, syntax, grooming

Explain how you tested your changes:
* I didn't run through any technical implementation testing

Checklist:

- [ ] Dependency versions are unchanged
  - Notify Velocity team if dependencies must change in CI
- [ ] Modified the current draft of release notes with details on what is completed or incomplete within this project
- [ ] Document code purpose, how to use it
  - Mention expected invariants, implicit constraints
- [ ] Tests were added for the new behavior
  - Document test purpose, significance of failures
  - Test names should reflect their purpose
- [ ] All tests pass (CI will check this if you didn't)
- [ ] Serialized types are in stable-versioned modules
- [ ] Does this close issues? List them

* Closes #0000

### CODE REVIEW

File: README-dev.md
Comment (barriebyron) [inline README-dev.md:18]: We can omit "currently"
https://developers.google.com/style/timeless-documentation

**Comment by barriebyron** [comment]
Hi Mina friends! We can remove "currently" from the repo description, too! Here are the suggested updates:

> Mina is a cryptocurrency with a constant-size blockchain, improving scaling while maintaining decentralization and security.

File: README-dev.md
Comment (nicc) [inline README-dev.md:123]:
```suggestion
To get all of the required opam dependencies, run:
```

File: README-dev.md
Comment (nicc) [inline README-dev.md:219]: Is this line break intentional?

**Review by nicc — COMMENTED** [review]
Looks good to me, with a few small editorial comments. But I don't know that I'm equipped to verify the integrity of the instructions. I will look into who else should review. Thanks Barrie!

File: README-dev.md
Comment (barriebyron) [inline README-dev.md:218]:
```suggestion
The `daemon.json` config file also contains bootstrap data that is specific to the network the node is trying to connect to and must be tailored specifically for a particular network. This file can also override some of the configuration options selected during compilation. The `daemon.json` file can be extracted from the Docker image
```

File: README-dev.md
Comment (barriebyron) [inline README-dev.md:219]:
```suggestion
that is dedicated to running a particular network. If it's not located in the `config` directory, it can be pointed to with `--config-file` option.
```

Comment (barriebyron) [reply README-dev.md:219]: @nicc  line breaks were already present in the files, tbh I don't know the impact or value

File: README-dev.md
Comment (Sventimir) [inline README-dev.md:133]: Not only must these libraries be available in the system, they also should be in appropriate versions, which is a major headache on most Linux systems, because various distibutions have different versions of libraries available in repositories at each time, depending on their update policy. Actually I tried to build Mina this way in the beginning, but after a week or so my setup broke down after a system update and I gave up, switching to Nix.

Although it has its own difficulties, I think Nix should be presented as the default way of building Mina, while this native build setup should be explicitly labeled as unstable (prone to random breakages) and it should only be proposed as an alternative to experienced Linux users.

File: README-dev.md
Comment (Sventimir) [inline README-dev.md:186]: One doesn't run the source code, but a compiled binary, which after a successful build appears in `_build/default/src/app/cli/src/mina.exe`.

File: README-dev.md
Comment (Sventimir) [inline README-dev.md:213]:
```suggestion
Additionally, you must provide a list of peers to connect to bootstrap the node.
```

File: README-dev.md
Comment (Sventimir) [inline README-dev.md:217]: It's worth mentioning that the `daemon.json` file first and foremost describes the genesis ledger, i.e. the initial state of the blockchain. It can also contain some protocol constants used throughout the network's lifetime, but these are all optional, as the node has default values for them compiled in. The only required parts of this config are the name of the network and the genesis ledger.

When setting up a fresh network, it is extremely important to add at least one account holding currency, to which *someone* posesses as matchiong private key, or the network will be devoid of funds. When connecting to an existing network, it is crucial that every node uses *the same* `daemon.json` file, or else there will be disagreaments on validity of blocks. Indeed, this config must be published along with the list of peers for every new network being created.

File: README-dev.md
Comment (Sventimir) [inline README-dev.md:232]: Mina CLI **is not** a **ZK** app (nor a "zkApp CLI"). On the contrary, it's an ordinary application, running the entire Mina blockchain, including (but not limited to) verifying zkApp execution proofs. Note that It **does not** execute any zkApps – these are being executed externally, in `nodejs` or in a user's browser, depending on their architecture. The blockchain is only concerned with verifying this execution was performed honestly.

File: README-dev.md
Comment (Sventimir) [inline README-dev.md:236]:
```suggestion
The Makefile contains placeholder targets for all the common tasks that need to be done and automatically knows how to use Docker. 
```
Although in principle the makefile  can be named anyhow, it is customary to name if `Makefile` (with capital "M"). This also the name used in Mina, so I'd use the capital letter to avoid confusion.

File: README-dev.md
Comment (Sventimir) [inline README-dev.md:253]: Since we're using ` ```shell ` everywhere, let's also change this to keep consistency and avoid confusion.

File: README.md
Comment (Sventimir) [inline README.md:11]: It might be worth mentioning here, there Mina's implementation of the [Rosetta API](https://www.rosetta-api.org/) offers a more stable (and often more useful) inferface for retrieving the blockchain's state. The downside of Rosetta is that it's run as a separate process, which also relies on an archive being connected to a node. The source code for the archive and Rosetta implementation are located in `src/app/archive` and `src/app/rosetta` respectively, although I heard there are plans to move them to separate repositories.

File: README.md
Comment (Sventimir) [inline README.md:52]: Is this still true? In MF we tried using Zenhub for a brief moment, but quickly switched back to working directly with Github issues and PRs. Maybe O(1) uses Zenhub, though.

**Review by Sventimir — CHANGES_REQUESTED** [review]
I like this editiorial effort, good job! :+1: I tried to correct some technical errors that I found.

Also I think in the developer's README, the Nix and Docker setups should be advertised more, while the native build setup should be explicitly labeled as difficult to maintain and only suggested to experienced users, who are not afraid to manually manage dependencies.

File: README-dev.md
Comment (barriebyron) [inline README-dev.md:6]:
```suggestion
- [Mina README](README.md)
```

File: README-dev.md
Comment (barriebyron) [reply README-dev.md:27]: @Sventimir  can you make direct updates with that suggestion? my updates did not introduce new content but just provided grooming for clarity
Since this change is out of scope for editorial grooming, we can skip your suggestion?

File: README-dev.md
Comment (barriebyron) [reply README-dev.md:133]: like my earlier suggestion, can you make changes to satisfy your technical change request? I value your feedback and feel your pain!

File: README-dev.md
Comment (barriebyron) [inline README-dev.md:186]:
```suggestion
The source code for the Mina node is located in `src/app/cli/`. After it is compiled, you can run the compiled binary like this:
```

File: README-dev.md
Comment (barriebyron) [inline README-dev.md:192]:
```suggestion

The results of a successful build appear in `_build/default/src/app/cli/src/mina.exe`.

The default configuration of the node depends on the build profile that is used during compilation. To connect to some networks, you need to compile the daemon with a specific profile.
```

File: README-dev.md
Comment (barriebyron) [reply README-dev.md:232]: Feedback like this @Sventimir  is so valuable, continue to teach me please! thank you

File: README-dev.md
Comment (barriebyron) [inline README-dev.md:253]:
```suggestion
```shell
```

File: README.md
Comment (barriebyron) [inline README.md:11]:
```suggestion
Mina is still under active development and APIs are evolving. If you build on the APIs, be aware that breaking changes can occur.

The Mina implementation of the Rosetta API offers a more stable and useful interface for retrieving the blockchain's state. Because Rosetta is run as a separate process, it relies on an archive being connected to a node. The source code for the archive and Rosetta implementation are in [src/app/archive](https://github.com/MinaProtocol/mina/tree/develop/src/app/archive) and [src/app/rosetta](https://github.com/MinaProtocol/mina/tree/develop/src/app/rosetta). Be sure to follow updates in the project if these resources are relocated. 
```

File: README-dev.md
Comment (barriebyron) [reply README-dev.md:220]: hi @Sventimir  can you provide a direct update to this PR? Or can we skip your suggestion so you can follow up in a separate effort?

File: README.md
Comment (barriebyron) [reply README.md:53]: I removed this section, iterative improvements in future can provide more context if required

File: README-dev.md
Comment (barriebyron) [reply README-dev.md]: I shared a video with you in the #mf-o1-docs channel on Slack

Comment (Sventimir) [reply README-dev.md]: As I mentioned on Slack, in my opinion the link should not only work in web browsers, but also in off-line editors that developers use to edit code (VS Code, Emacs and Vim seem to be the most popular at east among Mina engineers). Both web browsers and editors vary wildly in functionality, therefore the fact it works for you does not necessarily mean it'll work for everyone. Thus, it's best to stick to the filename exactly so that it works for the maximum number of setups people might use. Unfortunate as it is, this is the reality.

File: README-dev.md
Comment (Sventimir) [reply README-dev.md:27]: Yes, I'll prepare a separate PR.

File: README.md
Comment (Sventimir) [inline README.md:13]:
```suggestion
The Mina implementation of the Rosetta API offers a more stable and useful interface for retrieving the blockchain's state. Rosetta is run as a separate process and it relies on an archive being connected to a node. The source code for the archive and Rosetta implementation are in [src/app/archive](https://github.com/MinaProtocol/mina/tree/develop/src/app/archive) and [src/app/rosetta](https://github.com/MinaProtocol/mina/tree/develop/src/app/rosetta). Be sure to follow updates in the project if these resources are relocated. 
```
It's just a coincidence, no causal connection here.

**Review by Sventimir — APPROVED** [review]
Thank you for your changes! It looks good now.

**Comment by barriebyron** [comment]
@Sventimir  can you help me understand why CI is failing? Who can I tag as codeowner for review?

**Comment by barriebyron** [comment]
!ci-build-me

**Review by mitschabaude — APPROVED** [review]

File: README-dev.md
Comment (barriebyron) [inline README-dev.md:1]:
```suggestion
# Mina README-dev
```

**Comment by barriebyron** [comment]
!ci-build-me

**Comment by mitschabaude** [comment]
!ci-build-me

**Review by deepthiskumar — APPROVED** [review]

**Comment by mitschabaude** [comment]
!ci-build-me

**Review by bkase — COMMENTED** [review]
!approved-for-mainnet

**Comment by deepthiskumar** [comment]
!approved-for-mainnet

**Pull request merged**
